src/core/sock.c

```diff
@@ -122,6 +122,7 @@ int nn_sock_init (struct nn_sock *self, struct nn_socktype *socktype, int fd)
     self->linger = 1000;
     self->sndbuf = 128 * 1024;
     self->rcvbuf = 128 * 1024;
+    self->rcvmaxsize = 512 * 1024 * 1024;
```

Comment:
To be honest, this seems rather excessive.  I'd set by default to either -1 (unlimited), or something much smaller (1 MB).  Mangos sets this to 1MB by default.

512MB is so huge that it offers virtually no protection from abuse.